syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.singlelogin.account";
option java_outer_classname = "AccountProto";
option objc_class_prefix = "ACC";

package account;

service Account {
  rpc requestUserLogin  (LoginRequest)    returns (CodeReply) {}
  rpc requestUserSign   (SignRequest)     returns (CodeReply) {}
  rpc checkConnect      (ConnectRequest)  returns (CodeReply) {}
  rpc requestLogout     (LogoutRequest)   returns (CodeReply) {}
  rpc refreshToken      (RefreshRequest)   returns (CodeReply) {}

  rpc requestAddCategory  (AddCategoryRequest)    returns (CodeReply) {}
  rpc requestFetchCategory   (FetchCategoryRequest)     returns (CodeReply) {}
  rpc requestAddTodo      (AddTodoRequest)  returns (CodeReply) {}
  rpc requestUpdateTodo     (UpdateTodoRequest)   returns (CodeReply) {}
}

message LoginRequest {
  string account = 1;
  string password = 2;
}

message SignRequest {
  string account = 1;
  string password = 2;
}

message ConnectRequest {
  string token = 1;
}

message LogoutRequest {
  string token = 1;
}

message RefreshRequest {
  string token = 1;
  string refresh_token = 2;
}


message AddCategoryRequest {
  string token = 1;
  string title = 2;       // 名字
}

message FetchCategoryRequest {
  string token = 1;
}

message AddTodoRequest {
  string token = 1;
  string content = 2;   // todo 内容
  int32 cid = 3;       // 分类id
}

message UpdateTodoRequest {
  string token = 1;
  int32 tid = 2;       // todo项的id
  int32 status = 3;    // 状态
}

message CodeReply {
  int32 code = 1;
  string msg = 2;
  string data = 3;
}